From 1a10942708cd1ed2ac128c076d51f113a8db0974 Mon Sep 17 00:00:00 2001
From: Takuto Ikuta <tikuta@chromium.org>
Date: Mon, 7 May 2018 13:59:11 +0000
Subject: strings/char_traits.h: use HAS_FEATURE instead of __has_feature

This is for cl.exe build.

Change-Id: Icce6bbb29225d2f77d4b1105778aa26d9416c5e4
Reviewed-on: https://chromium-review.googlesource.com/1046425
Reviewed-by: Daniel Cheng <dcheng@chromium.org>
Commit-Queue: Takuto Ikuta <tikuta@chromium.org>
Cr-Commit-Position: refs/heads/master@{#556429}
---
 base/strings/char_traits.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/base/strings/char_traits.h b/base/strings/char_traits.h
index 6cfbad5af4d0..b193e216ccca 100644
--- a/base/strings/char_traits.h
+++ b/base/strings/char_traits.h
@@ -7,6 +7,8 @@
 
 #include <stddef.h>
 
+#include "base/compiler_specific.h"
+
 namespace base {
 
 // constexpr version of http://en.cppreference.com/w/cpp/string/char_traits.
@@ -61,7 +63,7 @@ struct CharTraits<char> {
 constexpr int CharTraits<char>::compare(const char* s1,
                                         const char* s2,
                                         size_t n) noexcept {
-#if __has_feature(cxx_constexpr_string_builtins)
+#if HAS_FEATURE(cxx_constexpr_string_builtins)
   return __builtin_memcmp(s1, s2, n);
 #else
   for (; n; --n, ++s1, ++s2) {
-- 
2.14.1

